drivers/backup/auth.go

```diff
@@ -690,7 +690,8 @@ func AddUser(userName, firstName, lastName, email, password string) error {
 		log.Errorf("%s: %v", fn, err)
 		return err
 	}
-	_, err = processHTTPRequest(method, reqURL, headers, strings.NewReader(string(userBytes)))
+	response, err := processHTTPRequest(method, reqURL, headers, strings.NewReader(string(userBytes)))
+	log.Infof("Response for user [%s] creation - %s", userName, string(response))
```

Comment:
Do we need to know response for all these commands when they are successful? Should we maybe just add these responses for the failures?

Reply:
Keycloak API never throws an error in case of failure. It returns a 2xx response even if the operation does succeed but returns the error as a string in the response. For us to know whether the API call was successful is to print out the response string and make further decisions based on that. For example, when you try to add a user to a group and it doesn't succeed, we do not get any error, but we get this error message in the response:

```
12:15:55  2023-02-02 06:45:55 +0000:[INFO log.go:#237]  Adding user to group response - {"error":"RESTEASY003210: Could not find resource for full path: [http://pxcentral-keycloak-http.px-backup.svc.cluster.local:80/auth/admin/realms/master/users//groups/7351abb4-2729-4e80-93b8-8ceca68e3259"}](http://pxcentral-keycloak-http.px-backup.svc.cluster.local/auth/admin/realms/master/users//groups/7351abb4-2729-4e80-93b8-8ceca68e3259%22%7D)
```

Reply:
how are you handling this failed responses?

Reply:
We are not sure what are the different type of error strings that we may get in the response. The Keycloak API documentation also does not mention about it.
As of now we are just going to print it out.
In the subsequent PR, we will check the error string and do 
`strings.Contains(string(response), "error message")`
If it contains error we will throw an error.

Reply:
We have to do this for all keycloak APIs that we are calling

Reply:
So if it doesn't return error, but you can see that the response has error why don't you do add a wrapper for API response to check if it has error or not and then handle it accordingly? I don't see a point to print response every time for no reason, if you not even getting anything out of that response, these responses are only useful to see when something is not working correctly or throwing an error, otherwise lets not print them, too much of not useful log.

Reply:
As I said, we do not have an official documentation which can let us know what are the different types of error strings that we can expect when there are failures. I agree that it will add a lot of unwanted log entries, but this is the first step in identifying what are the error messages that we can expect. In the subsequent PRs we will handle the error based on the response.
Some of these are legacy functions written long back, we are yet to figure out and apply the error handling logic to all keycloak related APIs which is in our TODO list.

Reply:
If you think we should not have these log entries at all, then I can remove them and add them later when we have everything figured out.
Anyway, adding log entries was not the main object of this PR. It was to add the cleanup functions.